bundles/org.openhab.binding.radiobrowser/src/main/java/org/openhab/binding/radiobrowser/internal/RadioBrowserHandler.java

```diff
@@ -0,0 +1,156 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.radiobrowser.internal;
+
+import static org.openhab.binding.radiobrowser.internal.RadioBrowserBindingConstants.*;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.radiobrowser.internal.api.ApiException;
+import org.openhab.binding.radiobrowser.internal.api.RadioBrowserApi;
+import org.openhab.binding.radiobrowser.internal.api.RadioBrowserJson.Country;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RadioBrowserHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Matthew Skinner - Initial contribution
+ */
+@NonNullByDefault
+public class RadioBrowserHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final LocaleProvider localeProvider;
+    public final RadioBrowserStateDescriptionProvider stateDescriptionProvider;
+    public RadioBrowserConfiguration config = new RadioBrowserConfiguration();
+    private RadioBrowserApi radioBrowserApi;
+    private @Nullable ScheduledFuture<?> reconnectFuture = null;
+
+    public RadioBrowserHandler(Thing thing, HttpClient httpClient,
+            RadioBrowserStateDescriptionProvider stateDescriptionProvider, LocaleProvider localeProvider) {
+        super(thing);
+        this.localeProvider = localeProvider;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        radioBrowserApi = new RadioBrowserApi(this, httpClient);
+    }
+
+    public void setChannelState(String channelToUpdate, State valueOf) {
+        updateState(channelToUpdate, valueOf);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                switch (channelUID.getId()) {
+                    case CHANNEL_STATION:
+                        radioBrowserApi.updateStations();
+                        return;
+                }
+            } else if (command instanceof StringType) {
```

Comment:
```suggestion
            } else if (command instanceof StringType commandAsString) {
```
Minor improvement regarding java 17 pattern matching. 
If this is change,d all the `command.toString()` can be changed to `commandAsString`

Reply:
I did try that before making the PR, and I tried again now to get the error. Eclipse is saying that the StringType can not be cast to String. Is using commandAsString.toString() better in any way or I am missing something?

Reply:
Thje StringType seems to work different then i expected. Forget my comment.